server/reload_test.go

```diff
@@ -1787,6 +1788,183 @@ func TestConfigReloadClusterName(t *testing.T) {
 	}
 }
 
+func TestConfigReloadClusterMembershipReload(t *testing.T) {
+	s1, _, conf1 := runReloadServerWithContent(t, []byte(`
```

Comment:
Add in leafnodes as well since that is what we really need to test.

Reply:
Added tests that include the leafnodes as well.